ui/library/src/utils/scroll.ts

```diff
@@ -78,13 +71,50 @@ export function scrollToPosition(
     leftPixels = (width * (PDF_HEIGHT_POINTS - bottomPoints)) / PDF_HEIGHT_POINTS;
   }
 
-  targetDiv.scrollTo({
-    top: Math.floor(heightWithMargins * pageIndex + marginTopPixels + (height - bottomPixels)),
+  const pageId = generatePageIdFromIndex(pageIndex);
+
+  const pageIdElement = document.getElementById(pageId);
+
+  if (!pageIdElement) {
+    return;
+  }
+
+  const parentElement = getScrollParent(pageIdElement);
+
+  if (!parentElement) {
+    return;
+  }
+
+  parentElement.scrollTo({
+    top: calculateTop(heightWithMargins, pageIndex, marginTopPixels, height, bottomPixels),
     left: Math.floor(leftPixels),
     behavior: 'smooth',
   });
 }
 
+export function getScrollParent(node: HTMLElement): HTMLElement | null {
```

Comment:
```suggestion
export function getScrollParent(node: HTMLElement): Nullable<HTMLElement> {
```

Reply:
@pauls-ai2 there is no Nullable in pdf-component xD